### PR TITLE
Fix compilation with Cython

### LINKED ISSUE #1008: Cython support is broken.
Running `python setup.py install --with-cython` yield an error about `visitor_expr.pyx` not existing.

It turns out that the list of files-to-Cythonize in `setup.py` is not up-to-date.

### PR DESCRIPTION
**Pull request by basilfx**
## Fixes 
#1008 

## Summary/Motivation:
This re-enabled Cython compilation using latest master. But also the latest PyPI version is broken.

## Changes proposed in this PR:
- Remove the file-to-Cythonize that does not exist.
- Silence warnings about language version.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by basilfx** [comment]
The missing file was (I guess) renamed from `visitor_expr.py` to `visitor.py`. However, this fails due to missing classes, which cannot be imported due to cyclic imports.

One way to fix this, is to provide `core/expr/visitor.pxd` as follows:

```Cython
from .numberic_expr import (
    clone_counter
    LinearExpression, 
    MonomialTermExpression,
    NPV_expression_types
)
```

But I can understand this makes the code base too opinionated.

File: setup.py [hunk 1]
```diff
@@ -90,7 +90,6 @@ def get_version():
             "pyomo/core/expr/numvalue.pyx",
             "pyomo/core/expr/numeric_expr.pyx",
             "pyomo/core/expr/logical_expr.pyx",
-            "pyomo/core/expr/visitor_expr.pyx",
```
Comment (jsiirola) [inline setup.py]: This should not be removed, but instead be changed to `"pyomo/core/expr/visitor.pyx"`

File: setup.py [hunk 2]
```diff
@@ -100,7 +99,8 @@ def get_version():
         ]
         for f in files:
             shutil.copyfile(f[:-1], f)
-        ext_modules = cythonize(files)
+        ext_modules = cythonize(files, compiler_directives={
+            "language_level": 3 if sys.version_info >= (3, ) else 2})
```
Comment (jsiirola) [inline setup.py]: This is a good fix.

File: setup.py (hunk 1 above)
Comment (basilfx) [reply setup.py]: Please see my second comment.

Comment (jsiirola) [reply setup.py]: Doh.  You are, of course, correct.  We should probably leave it in as a comment, though: the visitor file has all the routines for walking expression trees and a good chunk of the benefit of cython will come from cythonizing that file, so we should try and get the circular dependence resolved soon.

**Comment by ghackebeil** [comment]
Can we add a build to Travis that actually tests this code? I'd be okay with just testing it with Python 3.7. Maybe even running a more lightweight set of tests (e.g., 'smoke').

**Comment by ghackebeil** [comment]
Okay. That's fine with me.

**Pull request merged**